src/wsdiscovery.ts

```diff
@@ -27,32 +28,52 @@ export default class WsDiscovery extends EventEmitter {
             `WsDiscovery initilized with the following options: ${JSON.stringify(this.opts)}`,
             WsDiscovery.name
         );
+
+        this.bindSocketInterval = setInterval(() => this.bindSocket(), 1000);
         this.bindSocket();
     }
 
     bindSocket(): void {
-        this.socket = this.opts.socket || dgram.createSocket('udp4');
-        this.socket.bind(() => {
-            const map = this.findL1HostInterface();
-            if ((map.ip && map.interface) || this.opts.targetInterfaceAddr) {
-                const targetInterfaceAddr = this.opts.targetInterfaceAddr || map.ip;
-                this.socket.setMulticastInterface(targetInterfaceAddr);
-                Logger.debug(
-                    `Set default outgoing multicast interface of the socket to interface with address ${targetInterfaceAddr}`,
-                    WsDiscovery.name
-                );
-            }
-            if (!this.interfaceWatcher) {
-                Logger.debug(
-                    `Initiating interface watch for ${map.interface || 'default'}`,
-                    WsDiscovery.name
-                );
-                this.watchInterface(map.interface || 'default');
-            }
-        });
+        const map = this.findHostInterface();
+        const probeEntireNetwork =
+            this.opts.probeEntireNetwork || process.env.HUDDLY_WSDD_PROBE_ENTIRE_NETWORK;
+
+        if (!map.ip && !probeEntireNetwork) {
+            return;
+        }
+
+        const targetInterfaceAddr = map.ip;
+        if (probeEntireNetwork) {
+            this.socket = this.opts.socket || dgram.createSocket('udp4');
+            this.socket.bind();
+            Logger.debug(`Discovery probe messages bound to entire network`, WsDiscovery.name);
+            this.ifStateConnected = true;
+        } else if (targetInterfaceAddr) {
+            this.socket = this.opts.socket || dgram.createSocket('udp4');
+            this.socket.bind({ address: targetInterfaceAddr });
+            Logger.debug(
+                `Discovery probe messages bound to interface with ip: ${targetInterfaceAddr}`,
+                WsDiscovery.name
+            );
+            this.ifStateConnected = true;
+        }
+
         this.socket.on('error', err => {
             this.emit('ERROR', err);
         });
+
+        if (!this.interfaceWatcher) {
+            Logger.debug(
+                `Initiating interface watch for ${map.interface || 'default'}`,
+                WsDiscovery.name
+            );
+            this.watchInterface(map.interface || 'default');
+        }
+
+        if (this.bindSocketInterval) {
+            clearInterval(this.bindSocketInterval);
+            this.bindSocketInterval = undefined;
+        }
```

Comment:
Could you please elaborate on this? What happens if you plug a second base after the `bindSocketInterval` has been cleared?

Reply:
Also have you tested by doing several iterations of plug->unplug->plug->unplug of the BASE with L1 connected? How does the discovery behave in that case? Also important to test plugging in and out only the L1 from BASE